digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/ProjectDeltaCalculator.scala

```diff
@@ -528,7 +528,30 @@ object ProjectDeltaCalculator {
     })
   }
 
-  def calc_parts(twoTrackOldAddressRoadParts: Map[(Long, Long), Iterable[Seq[(RoadwaySection, String)]]]): Map[Seq[RoadwaySection], Seq[RoadwaySection]] = {
+  /** Create grouping for old address parts.
+   * Utility function for two track terminated change table address matching.
+   *
+   * @param unChanged_roadway_sections   Roadway change mapping for Unchanged roadway sections
+   * @param transferred_roadway_sections Roadway change mapping for Transferred roadway sections
+   * @param terminated_roadway_sections  Roadway change mapping for Terminated roadway sections
+   * @return RoadNumber, RoadPartNumber, Track grouping with `other, terminated´ labelings
+   */
+  def createTwoTrackOldAddressRoadParts(unChanged_roadway_sections: Iterable[(RoadwaySection, RoadwaySection)],transferred_roadway_sections: Iterable[(RoadwaySection, RoadwaySection)], terminated_roadway_sections: ChangeTableRows2): Map[(Long, Long), Iterable[Seq[(RoadwaySection, String)]]] = {
+    ((unChanged_roadway_sections ++ transferred_roadway_sections).map(roadwaySection => (roadwaySection._2, "other")).toSeq
+     ++
+     terminated_roadway_sections.adjustedSections.map(roadwaySection => (roadwaySection, "terminated")).toSeq)
+    .filterNot(_._1.track == Track.Combined)
+    .sortBy(_._1.startMAddr)
+    .groupBy(p => (p._1.roadNumber, p._1.roadPartNumberStart))
+    .mapValues(p => p.groupBy(_._1.track).values)
+  }
```

Comment:
Tämä ei ehkä ollut se mitä hain takaa. 😄
Nyt tämä ketjutus on siivottu funktion taakse piiloon mutta tämä pitää silti lukea läpi ja ymmärtää ja nyt se on aivan liian sekava (= liian monta ketjutusta peräkkäin, jolloin luettavuus katoaa.)

Tämän logiikan ei tarvitse mahtua viidelle riville. Eli haluasin että ketjutus pilkotaan muuttujien avulla pienempiin osiin.